src/bootchooser.c

```diff
@@ -284,6 +286,57 @@ static gboolean grub_set_primary(RaucSlot *slot) {
 	return res;
 }
 
+static gboolean uboot_env_get(const gchar *key, GString **value) {
+	GSubprocess *sub;
+	GError *error = NULL;
+	GBytes *stdout_buf = NULL;
+	const char *data;
+	gsize offset;
+	gsize size;
+	gboolean res = FALSE;
+	gint ret;
+
+	g_assert_nonnull(key);
+
+	sub = g_subprocess_new(G_SUBPROCESS_FLAGS_STDOUT_PIPE, &error,
+			       UBOOT_FWGETENV_NAME, key, NULL);
+	if (!sub) {
+		g_warning("starting " UBOOT_FWGETENV_NAME " failed: %s",
+			  error->message);
+		g_clear_error(&error);
+		goto out;
+	}
+
+	res = g_subprocess_communicate(sub, NULL, NULL, &stdout_buf, NULL, &error);
+	if (!res) {
+		g_warning(UBOOT_FWGETENV_NAME " communication failed: %s", error->message);
+		g_clear_error(&error);
+		goto out;
+	}
+
+	res = g_subprocess_get_if_exited(sub);
+	if (!res) {
+		g_warning(UBOOT_FWGETENV_NAME " did not exit normally");
+		goto out;
+	}
+
+	ret = g_subprocess_get_exit_status(sub);
+	if (ret != 0) {
+		g_warning(UBOOT_FWGETENV_NAME " failed with exit code: %i", ret);
```

Comment:
We need to have a `res = FALSE` here, otherwise `uboot_env_get()` would return TRUE, as previously set by `res = g_subprocess_get_if_exited(sub);`

Reply:
Pushed an update...